### PR TITLE
AzureMonitor: Add ResourceGroups template variable

### PR DESCRIPTION
**Pull request by andresmgot**
**What this PR does / why we need it**:

Another function for https://github.com/grafana/grafana/issues/50231

Screenshot from 2022-07-13 11-58-20

**Which issue(s) this PR fixes**:

Ref https://github.com/grafana/grafana/issues/50231

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/components/VariableEditor/VariableEditor.tsx [hunk 1]
```diff
@@ -51,6 +71,17 @@ const VariableEditor = (props: Props) => {
...
   };
 
+  const onChangeSubscription = (selectableValue: SelectableValue) => {
+    if (selectableValue.value) {
+      const newQuery = {
+        ...query,
+        subscription: selectableValue.value,
+      };
+      setQuery(newQuery);
+      props.onChange(newQuery);
```
Comment (aangelisc) [inline public/app/plugins/datasource/grafana-azure-monitor-datasource/components/VariableEditor/VariableEditor.tsx:81]: Would it be a better pattern here to `useEffect(() => props.onChange(newQuery), [newQuery])`?

**Review by aangelisc — APPROVED** [review]
LGTM! Just one question inline 😊

**Review by aangelisc — APPROVED** [review]

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/components/VariableEditor/VariableEditor.tsx (hunk 1 above)
Comment (aangelisc) [reply public/app/plugins/datasource/grafana-azure-monitor-datasource/components/VariableEditor/VariableEditor.tsx]: Yup makes sense to me 😊

**Review by asimpson — APPROVED** [review]
LGTM :+1:

**Pull request merged**